### PR TITLE
 add gzip support, add base64 logs

### PR DESCRIPTION
**Pull request by xmanjack**
- add gzip support 
- add base64 logs

### CODE REVIEW

**Comment by xmanjack** [comment]
@riot-jayb 
@jeffizhungry Logs finally workings!, no leading weird char anymore

**Comment by xmanjack** [comment]
time boxed for 5 days.

File: conf/production/app_prod.conf [hunk 1]
```diff
@@ -2,6 +2,7 @@ appname = docker-logs-agent
 httpport = 7001
 runmode = prod
 autorender = false
-copyrequestbody = true
+copyrequestbody = false
 EnableDocs = false
 docker_api_version = 1.39
+EnableGzip = true
```
Comment (jeffizhungry) [inline conf/production/app_prod.conf:8]: why are these config vars mixed case? (both staring with Uppercase and lowercase)

File: models/container.go [hunk 2]
```diff
@@ -2,49 +2,71 @@ package models
...
+	ContainerLog       containerLog       `json:"containerLog"`
+	Base64ContainerLog base64ContainerLog `json:"base64ContainerLog"`
+}
 
-type Container struct {
-	containerID string
-	Logs        string
+type containerLog struct {
+	ContainerID string `json:"containerID"`
+	Logs        string `json:"Logs"`
```
Comment (jeffizhungry) [inline models/container.go:21]: you should be consistent and lowercase this

File: models/container.go [hunk 3]
```diff
@@ -2,49 +2,71 @@ package models
...
 
-func init() {
-	Containers = make(map[string]*Container)
+type base64ContainerLog struct {
+	Base64containerID string `json:"base64containerID"`
+	Base64Logs        []byte `json:"base64Logs"`
 }
 
-func GetLog(containerID string, tail string) (string, error) {
+func GetLog(containerID string, tail string) (Logs, error) {
```
Comment (jeffizhungry) [inline models/container.go:29]: TIP: for larger projects, you should decouple your `models` package (where your types are defined) from your business logic (how the models are used). The reason being, that there will be a lot of packages in your project that need to interact with your data models, but they don't all need to know about how to fetch, persist or process them.

File: models/container.go [hunk 4]
```diff
@@ -2,49 +2,71 @@ package models
...
-	containerID string
-	Logs        string
+type containerLog struct {
+	ContainerID string `json:"containerID"`
+	Logs        string `json:"Logs"`
 }
 
-func init() {
-	Containers = make(map[string]*Container)
+type base64ContainerLog struct {
```
Comment (jeffizhungry) [inline models/container.go:24]: why do we need base64 logs? can you provide me some context?

**Review by jeffizhungry — COMMENTED** [review]
Left some comments

**Pull request merged**